### PR TITLE
Added README.md and api.md for bmp server

### PR DESCRIPTION
**Pull request by zhanggbj**
Hi @maximilien,

This is ready for review. Thanks.

Fixed issue #8 and issue #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
-# bosh-softlayer-baremetal-server
-A server to manage SoftLayer baremetals servers and a BOSH release to BOSHify the deployment and management of the server.
+bosh-softlayer-baremetal-server
+===============================
+
+A server to create, provision and manage SoftLayer baremetals servers.
+
+## Setup
+The server could be deployed by BOSH with a release as below. The release contains three jobs named baremetal-provision-server, xcat-server and redis. Job baremetal-provision-server and xcat-server should always be on the same node.
```
Comment (maximilien) [inline README.md]: Use the back quote `like this` when naming things so they stand out, e.g., `baremetal-provision-server`, `xcat-server`, and `redis`.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
+https://github.com/cloudfoundry-community/bosh-softlayer-baremetal-server-releas
```
Comment (maximilien) [inline README.md]: Fix this, missing last "e":
```
https://github.com/cloudfoundry-community/bosh-softlayer-baremetal-server-release
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,56 @@
...
+
+This is an example deployment yaml file.
+https://github.com/cloudfoundry-community/bosh-softlayer-baremetal-server-release/docs/baremetal-provision-server.yml
```
Comment (maximilien) [inline README.md]: Add it as a link, e.g.,

```
[https://github.com/cloudfoundry-community/bosh-softlayer-baremetal-server-release/docs/baremetal-provision-server.yml](https://github.com/cloudfoundry-community/bosh-softlayer-baremetal-server-release/docs/baremetal-provision-server.yml)
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,56 @@
...
+
+## Usage
+Once deployment is completed, you could login the server and take check.
```
Comment (maximilien) [inline README.md]: Change "could" to "can"... the active voice in English makes things easier to read.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+Process 'baremetal-provision-worker-1' running
+Process 'baremetal-provision-worker-2' running
+Process 'baremetal-provision-worker-3' running
+Process 'baremetal-provision-worker-4' running
+Process 'baremetal-provision-worker-5' running
+Process 'baremetal-provision-worker-6' running
+System 'system_localhost'           running
+```
+If any job is down, you could go to `/var/vcap/sys/log` and take a look at the logs.
+
```
Comment (maximilien) [inline README.md]: Include some typical things to try if a job is down or what to look for. For instance, perhaps `$monit restart <job-name>` etc.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,56 @@
...
+The server exposes RESTful api which you could find details in `api.md` and you could use a CLI written in GO to leverage it. Firstly you need to set the target and login, then you could use the server to create and provision a baremetal.  Here is the link of CLI.
+https://github.com/cloudfoundry-community/bosh-softlayer-tools
```
Comment (maximilien) [inline README.md]: We should add a way to simply: `$go get https://github.com/cloudfoundry-community/bosh-softlayer-tools` and get the CLI. See how on my  [i18n4go](https://github.com/maximilien/i18n4go) project you can do: `$go get github.com/maximilien/i18n4go` and go will download and install the CLI. I think you will need to add a file at the project root that contains `main` function like I do in `i18n4go`.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,56 @@
...
+>
+>$ bmp login -u admin -p admin
+>{"status":200,"data":null}
+>
+>$ bmp status
+>BMP server info
+ name:    Bluemix Provision Server
+ version: 0.1
+
+When to create a baremetal on Softlayer, you need to create a `deployment.yml` firstly and specify the your required configuration like core or mem.
```
Comment (maximilien) [inline README.md]: Be specific, `core` and `mem` are not actual things you specify, correct? Only list things that user actually specify as you risk confusing reader.

File: docs/api.md [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+curl -X GET "http://{serverIP}:8080/info
+```
+**Response**
+Status
+`200 OK`
+
+Body
+```
+{
+    "name" : "fake-name",
```
Comment (maximilien) [inline docs/api.md]: Instead of `fake-name` please, except for username and password, include real examples.

**Comment by maximilien** [comment]
@zhanggbj this is a great start. Please address my comments and I'll merge. Thanks.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,56 @@
-# bosh-softlayer-baremetal-server
-A server to manage SoftLayer baremetals servers and a BOSH release to BOSHify the deployment and management of the server.
+bosh-softlayer-baremetal-server
```
Comment (maximilien) [inline README.md]: Make sure to include a link to the `api.md` in this doc.

**Comment by maximilien** [comment]
@zhanggbj can you please address and resubmit so I merge. I want to have some docs on this since the Diego team might be looking into baremetal for some of their tests.

Thanks much.

**Comment by zhanggbj** [comment]
Hi dr.max @maximilien,

All comments addressed. This is ready for review. Thank you.
Sorry for the late response. I didn't got enough time to update this PR due to attending an training last week.

**Comment by maximilien** [comment]
Cool, thanks. Merging. Cheers 👍

**Pull request merged**